control-plane/agents/src/bin/core/controller/scheduling/resources/mod.rs

```diff
@@ -101,6 +102,25 @@ impl PoolItemLister {
             None => vec![],
         }
     }
+    /// Get a list of pool items to resize the replicas that reside on them.
+    pub(crate) async fn list_for_resize(registry: &Registry, spec: &VolumeSpec) -> Vec<PoolItem> {
+        let pool_ids = registry.specs().volume_replica_pools(&spec.uuid);
+
+        let items = Self::nodes(registry)
+            .await
+            .iter()
+            .flat_map(|n| {
+                n.pool_wrappers()
+                    .iter()
+                    .filter(|p| registry.specs().pool(&p.id).is_ok() && pool_ids.contains(&p.id))
+                    .map(|p| PoolItem::new(n.clone(), p.clone(), None))
+                    .collect::<Vec<_>>()
+            })
+            .collect();
+
+        items
```

Comment:
nit: don't need variable, can just return

```suggestion
            .collect()
```